src/index.ts

```diff
@@ -11,6 +11,23 @@ export const keypairToSecretKeyJSON = (keypair: Keypair): string => {
   return JSON.stringify(Array.from(keypair.secretKey));
 };
 
+export const getCustomErrorMessage = (
+  possibleProgramErrors: Array<string>,
+  errorMessage: string,
+): string | null => {
+  const customErrorExpression =
+    /.*custom program error: 0x(?<errorNumber>[0-9abcdef]+)/;
+
+  let match = customErrorExpression.exec(errorMessage);
+  const errorNumberFound = match?.groups?.errorNumber;
+  if (!errorNumberFound) {
```

Comment:
Probably makes more sense to return the error instead of null

Reply:
The function is called `getCustomErrorMessage()` - if there's no custom error message, it should return nothing, as we could not find a custom error message. Returning the original error would be deceptive.